vocabulary-draft/continuous-integration-pipeline-events.md

```diff
@@ -0,0 +1,28 @@
+# Continuous Integration Events
+
+__Note:__ This is a work-in-progress draft version and is being worked on by members of the Events SIG. You are very welcome to join the work and the discussions!
+
+These events are related to **Continuous Integration(CI)** activities. CI usually include activities such as building, testing, packaging and releasing software artifacts. 
+
+The following events represent concrete Tasks that are associated with the execution of CI pipelines:
+
+- **Build Queued**: a Build task has been queued, this build process usually is in charge of producing a binary from source code
+- **Build Started**: a Build task has started 
+- **Build Finished**: a Build task has finished, the event will contain the finished status, success, error or failure
+
+The following Test events are defined in two separate categories **Test Case** and **Test Suite**. A **Test Case** is the smallest unit of testing that the user wants to track. A **Test Suite** is a collection of test case executions and/or other test suite executions. **Test Cases** executed, and Test Suites are for grouping purposes. For this reason, **Test Cases** can be queued. 
+
+- **Test Case Queued**: a Test task has been queued, and it is waiting to be started
+- **Test Case Started**: a Test task has started
+- **Test Case Finished**: a Test task has finished, the event will contain the finished status: success, error or failure
+- **Test Suite Started**: a Test Suite has started
```

Comment:
Could you make a practical example of a test case / test suite?
I think the terms might be confusing, as the first thing that came to mind for me for a test case is a single unit test or e2e test, and suite a collection of them, but I don't suppose we want tests at that level of granulaty?

Maybe a test suite would be pre-merge tests and test cases would be linting, build, unit, e2e?

Reply:
Let us say we have:

* Tests
  * Front-end test suite
	* 30 tests
  * Back-end test suite
	* 40 test
  * Performance test suite
	* 10 tests

If we wish to capture the hierarchy we can then have one Suite for all the tests which then have 3 suites. In each suite, the tests exist. Thus in this case we would send 4 suite events and 80 test case events.

Did this make any sense and answer your question @afrittoli ?